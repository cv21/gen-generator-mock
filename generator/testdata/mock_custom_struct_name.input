package somesvc

import "io"

type SomeService interface {
    SomeSliceGetter() []io.Reader
}