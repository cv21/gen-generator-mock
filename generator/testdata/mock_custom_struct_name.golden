// File generated by gen. DO NOT EDIT.
// Generator plugin github.com/cv21/gen-generator-mock v1.0.0.
package bla

import (
	mock "github.com/stretchr/testify/mock"
	"io"
)

// MockBlaBlaSomeService is an autogenerated mock type for the SomeService interface.
type MockBlaBlaSomeService struct {
	mock.Mock
}

// SomeSliceGetter provides a mock function for method SomeSliceGetter of interface SomeService.
func (_m *MockBlaBlaSomeService) SomeSliceGetter() []io.Reader {
	ret := _m.Called()

	var r0 []io.Reader
	if rf, ok := ret.Get(0).(func() []io.Reader); ok {
		r0 = rf()
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).([]io.Reader)
		}
	}

	return r0
}
